src/test/java/seedu/taskify/logic/parser/TagSearchCommandParserTest.java

```diff
@@ -0,0 +1,35 @@
+package seedu.taskify.logic.parser;
+
+import static seedu.taskify.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.taskify.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.taskify.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.taskify.logic.commands.TagSearchCommand;
+import seedu.taskify.model.task.TagContainsKeywordsPredicate;
+
+public class TagSearchCommandParserTest {
+
+    private TagSearchCommandParser parser = new TagSearchCommandParser();
+
+    @Test
+    public void parse_emptyArg_throwsParseException() {
+        assertParseFailure(parser, "     ", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                           TagSearchCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_validArgs_returnsTagSearchCommand() {
```

Comment:
could add a few more tests just to be safe, maybe something like "ModuleTutorial" for user input and predicate is {"Module", "Tutorial"}?

Reply:
Ok, will add! The tests I wrote so far follow the same format as the ones for the find command.

Reply:
The tests in `TagSearchCommandParserTest` checks if the parser splits on space correctly.  The test you mentioned is should be part of `TagContainsKeywordsPredicateTest`. I have added it there.